code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/ServiceContainerStatsLinkHandler.java

```diff
@@ -0,0 +1,117 @@
+package io.cattle.platform.host.stats.api;
+
+import io.cattle.platform.api.link.LinkHandler;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.docker.util.DockerUtils;
+import io.cattle.platform.host.model.HostApiAccess;
+import io.cattle.platform.host.service.HostApiService;
+import io.cattle.platform.host.stats.utils.StatsConstants;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.token.TokenService;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ServiceContainerStatsLinkHandler implements LinkHandler {
+
+    HostApiService hostApiService;
+    ObjectManager objectManager;
+    TokenService tokenService;
+
+    private static final String SERVICE_PATH = "/service/";
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "service" };
+    }
+
+    @Override
+    public boolean handles(String type, String id, String link, ApiRequest request) {
+        return StatsConstants.CONTAINER_STATS.equalsIgnoreCase(link);
+    }
+
+    @Override
+    public Object link(String name, Object obj, ApiRequest request) throws IOException {
+        Service service = (Service) obj;
+
+        List<Instance> serviceInstances = objectManager.mappedChildren(service, Instance.class);
+
+        if (service == null) {
+            return null;
+        }
+
+        String metaUrl = null;
+        StatsAccess meta = new StatsAccess();
+        List<StatsAccess> serviceStatsQuery = new ArrayList<>();
+        for (Instance instance : serviceInstances) {
```

Comment:
What is the reason for including Stopped state to the list of valid states for the stats? If there is any stats available for such containers, we should also include other states - rather than Removed one - to the list